src/main/java/com/example/playdocker/controllers/MessageController.java

```diff
@@ -0,0 +1,53 @@
+package com.example.playdocker.controllers;
+
+import com.example.playdocker.Utils.MessageUtils;
+import com.example.playdocker.clients.GoogleClient;
+import com.example.playdocker.models.Message;
+import com.example.playdocker.models.Score;
+import com.example.playdocker.services.MessageService;
+import org.springframework.web.bind.annotation.*;
+import java.util.ArrayList;
+
+import static com.example.playdocker.services.MessageService.findUrls;
+
+@RestController
+public class MessageController {
+
+    @PostMapping("/receiveMessage")//Funkcja przyjmująca request typu POST.
+    public String receiveMessage(@RequestBody Message message){
+
+        String apiKey = MessageUtils.getApiKeyFromProperties();//Pobieranie klucza API z propertiesów
+        ArrayList<Score> scores = new ArrayList<Score>();//Lista wyników testowanych URLi zwrócona przez API google
+        ArrayList<String> listOfUrls = findUrls(message.getContent());//Tworzenie listy z URLi wyszukanych w wiadomości
+        for (int i = 0; i < listOfUrls.size(); i++) {//Pętla pobierająca wyniki testów dla każdego z URLi i dodaje je do listy wyników
+            Score score = GoogleClient.getScoreForUrl(listOfUrls.get(i), apiKey);
+            scores.add(score);
+        }
+        boolean isMessageDangerous = MessageService.checkForDangerousLinksInArray(scores);//Sprawdzenie, czy na liście wyników któryś z URLi nie jest niebezpieczny
+        MessageService.saveMessage(message);//Funkcja zapisująca wiadomość do bazy danych
+        if(isMessageDangerous) {
+            return "Wiadomość może być potencjalnie niebezpieczna";
+        } else {
+            return "Wiadomość jest bezpieczna";
+        }
```

Comment:
- Takie elementy warto oddzielać od controllera, to jest pewnego logika biznesowa, moze warto umieścić ją w innym miejscu? Jakim? 
- Jak zwróciłbyś nie tylko wiadomość `"Wiadomość jest bezpieczna";` ale również informację dla serwisu, który korzysta z Twojego rozwiązania z kodem odpowiedzi http? Jaki kod byłby tu odpowiedni? 
- Jak obsługujesz błędy, które mogą wydarzyć się po drodze?